extensions/wikia/Follow/FollowHelper.class.php

```diff
@@ -39,88 +36,39 @@ static public function watchCategories( $categoryInserts, $categoryDeletes, $tit
 
 		self::emailNotification( $title, $queryIn, NS_CATEGORY, $wgUser, $action, wfMsg( 'follow-categoryadd-summary' ) );
 
-		wfProfileOut( __METHOD__ );
 		return true;
 	}
 
 	/**
 	 * emailNotification -- sent Notification for all related article
 	 *
-	 * @static
-	 * @access public
-	 *
-	 *
 	 * @param Title $childTitle
-	 * @param $list
-	 * @param $namespace
+	 * @param array $list
+	 * @param string $namespace
 	 * @param User $user
-	 * @param $action
-	 * @param $message
+	 * @param string $action
+	 * @param string $message
 	 *
 	 * @throws MWException
 	 */
 	public static function emailNotification( $childTitle, $list, $namespace, $user, $action, $message ) {
-		wfProfileIn( __METHOD__ );
 
 		if ( count( $list ) < 1 ) {
-			wfProfileOut( __METHOD__ );
 			return;
 		}
 
-		$dbw = wfGetDB( DB_SLAVE );
-		$queryIn = array();
-		foreach ( $list as $value ) {
-			$queryIn[] = $dbw->addQuotes( $value ) ;
-		}
-
-		/* Wikia change begin - @author: wladek */
-		/* RT#55604: Add a timeout to the watchlist email block */
+		$watcherSQL = self::getWatcherSQL( $user->getId(), $namespace, $list );
+		$watcherSets = self::getWatcherSets( $watcherSQL );
 
-	 	global $wgEnableWatchlistNotificationTimeout, $wgWatchlistNotificationTimeout;
-
-		$now = wfTimestampNow();
-		if ( !empty( $wgEnableWatchlistNotificationTimeout ) && isset( $wgWatchlistNotificationTimeout ) ) { // not using !empty() to allow setting integer value 0
-			$blockTimeout = wfTimestamp( TS_MW, wfTimestamp( TS_UNIX, $now ) - intval( $wgWatchlistNotificationTimeout ) );
-			$notificationTimeoutSql = "(wl_notificationtimestamp IS NULL OR wl_notificationtimestamp < '$blockTimeout')";
-		} else {
-			$notificationTimeoutSql = "wl_notificationtimestamp IS NULL";
-		}
-
-		if ( $action == self::LOG_ACTION_BLOG_POST ) {
-			$notificationTimeoutSql = "1";
```

Comment:
The developer added this so that they wouldn't have to decide if they should include `$notificationTimeoutSql` in the select.  Its always included, and in this case where they don't have a timeout, this looks like:
```php
SELECT ...
...
AND 1
```
So no, this is not logic that needs to be retained.